src/authentication/components/StamboekInput.tsx

```diff
@@ -0,0 +1,171 @@
+						type="inline-link"
+					/>{' '}
+					toch al toegang krijgen.
+				</span>
+			),
+			status: ToastType.DANGER,
+		},
+		VALID_FORMAT: {
+			message: 'Bezig met valideren',
+			status: ToastType.SPINNER,
+		},
+		VALID: {
+			message: 'Het stamboek nummer is geldig',
+			status: ToastType.SUCCESS,
+		},
+		ALREADY_IN_USE: {
+			message: (
+				<span>
+					Dit stamboek nummer is reeds in gebruik,{' '}
+					<Button
+						label="contacteer de helpdesk"
+						onClick={() => redirectToClientPage(APP_PATH.MANUAL_ACCESS_REQUEST, history)}
+						type="inline-link"
+					/>
+					.
+				</span>
+			),
+			status: ToastType.SUCCESS,
+		},
+		SERVER_ERROR: {
+			message:
+				'Er ging iets mis bij het valideren van het stamboek nummer. Probeer later eens opnieuw.',
+			status: ToastType.DANGER,
+		},
+	};
+
+	const setStamboekNumber = async (rawStamboekNumber: string) => {
+		try {
+			setRawStamboekNumber(rawStamboekNumber);
+			const cleanedStamboekNumber = rawStamboekNumber.replace(/[^0-9-]+/, '');
+			// Check if stamboek number is incomplete
+			// eg: 3256
+			// or: 43457876543-34
+			if (/^[0-9]{0,10}$/g.test(cleanedStamboekNumber)) {
+				setStamboekValidationStatus('INCOMPLETE');
+				return;
+			}
+			if (/^[0-9]{11}(-[0-9]*)?$/g.test(cleanedStamboekNumber)) {
+				const stamboekNumber = cleanedStamboekNumber.substring(0, 11);
+				setStamboekValidationStatus('VALID_FORMAT');
+				const validationStatus: StamboekValidationStatuses = await verifyStamboekNumber(
+					stamboekNumber
+				);
+				if (validationStatus === 'VALID') {
+					onChange(stamboekNumber);
+					setStamboekValidationStatus('VALID');
+				} else if (validationStatus === 'ALREADY_IN_USE') {
+					setStamboekValidationStatus('ALREADY_IN_USE');
+				} else {
+					// 'INVALID' server response
+					setStamboekValidationStatus('INVALID_NUMBER');
+				}
+			} else {
+				setStamboekValidationStatus('INVALID_FORMAT');
+			}
+		} catch (err) {
+			setStamboekValidationStatus('SERVER_ERROR');
+		}
+	};
+
+	return (
+		<Spacer className="m-stamboek-input" margin={['bottom-large']}>
+			<TextInput
+				placeholder="00000000000-000000"
+				value={rawStamboekNumber}
+				onChange={setStamboekNumber}
+			/>
+			<Tooltip position="bottom" contentClassName="m-stamboek-tooltip">
+				<TooltipTrigger>
+					<span>
+						<Icon className="a-info-icon" name="info" size="small" />
+					</span>
+				</TooltipTrigger>
+				<TooltipContent>
+					<Spacer margin={'small'}>
+						<Spacer margin="bottom-small">
+							<span>Je stamboek nummer staat op je lerarenkaart</span>
+						</Spacer>
+						<img
+							alt="Voorbeeld leeraren kaart"
+							className="a-stamboek-image"
+							src="/images/leerkrachten-kaart-voorbeeld-nummer.png"
+						/>
+					</Spacer>
+				</TooltipContent>
+			</Tooltip>
+			<Spacer margin="top-small">
+				{STAMBOEK_MESSAGES[stamboekValidationStatus].status && (
+					<Alert type={STAMBOEK_MESSAGES[stamboekValidationStatus].status}>
+						<div>
```

Comment:
Is dit gewoon omdat er een HTML element in Alert moet zitten? Indien ja, werkt dit mogelijks met een fragment?